client/settings/settings-tab-panel/test/index.test.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import { UPESettingsTabPanel } from '../';
+
+it( 'should render two tabs when mounted', () => {
+	render( <UPESettingsTabPanel /> );
+	expect(
+		screen.getByRole( 'tab', { name: /Payment Methods/i } )
+	).toBeInTheDocument();
+	expect(
+		screen.getByRole( 'tab', { name: /Settings/i } )
+	).toBeInTheDocument();
+} );
+
+it( 'should change tabs when clicking on them', () => {
+	render( <UPESettingsTabPanel /> );
+	const settingsButton = screen.getByRole( 'tab', { name: /Settings/i } );
+	const methodsButton = screen.getByRole( 'tab', {
+		name: /Payment Methods/i,
+	} );
+	userEvent.click( settingsButton );
+	expect( settingsButton ).toHaveClass( 'is-active' );
```

Comment:
Expecting on class names can be brittle, what do you think about expecting on the content being rendered instead?

Reply:
I don't like doing this either generally, but I wasn't really sure _what_ content would be best to check since the underlying components haven't been created yet. There isn't at this moment any real content to check. 🤔

Reply:
The content of `PaymentMethodsPanel` is "Payment Methods". You could rename it to "Payment methods content" (so that it doesn't clash with the tab's name) and then do `expect( screen.queryByText( 'Payment methods content' ).toBeInTheDocument()`.
Or on the `PaymentMethodsPanel` you could render a `div` with a `data-testid="payment-methods-content"` and then do `expect( screen.queryByTestId( 'payment-methods-content' ).toBeInTheDocument()`.

Reply:
`expect( screen.queryByText( 'Payment methods content' ).toBeInTheDocument()`

The problem is that when someone actually implements this component, this test _definitely_ breaks(that text will be removed) and the dev will have to update the test. 
So my thinking was, "what can I test that won't immediately break when someone implements this component" and then when we have a PR for the PaymentMethods component it can contain a relevant test. 🤔

Reply:
> The problem is that when someone actually implements this component, this test definitely breaks(that text will be removed) and the dev will have to update the test. Or is that your intention here?

Yeah, that's kinda my intention and I think it should be expected. But with the `data-testid` approach instead we don't even occur into the issue.